syntax = "proto3";

package battleground;

import "character.proto";

service BattlegroundService {
    rpc Fighting(BattlegroundRequest) returns (BattlegroundResponse);
}

message BattlegroundRequest {
    character.Character first_player = 1;
    character.Character second_player = 2;
}

message BattlegroundResponse {
    character.Character winner = 1;
    repeated BattleLog battle_log = 2;
}

message BattleLog {
    character.Character attacker = 1;
    character.Character defender = 2;
    int32 damage = 3;
    int32 defence_health = 4;
    repeated character.Skills skills = 5;
}
